### PR TITLE
Add logout function and an API call in SessionValid for session testing

### PR DESCRIPTION
**Pull request by David-T-White**
Added a Logout call to the openAPI spec generation code to gracefully end API sessions. 

Also added a call to 'show system' in the SessionValid function as an additional test for validity.

### CODE REVIEW

File: pkg/regression/v2_login.go [hunk 1]
```diff
@@ -32,5 +32,17 @@ var _ = DescribeRegression("Login Testing (v2)", func(tc *TestContext) {
 			Expect(valid).To(Equal(true))
 		})
 
+		It("should be able to log out of the storage controller", func() {
+			err := client.Logout()
+			logger := klog.FromContext(tc.Config.Ctx)
+			logger.V(3).Info("Login", "ip", client.Addr, "username", client.Username, "err", err)
```
Comment (jskazinski) [inline pkg/regression/v2_login.go:38]:
```suggestion
			logger.V(3).Info("Logout", "ip", client.Addr, "username", client.Username, "err", err)
```

File: pkg/api/volumes.go [hunk 2]
```diff
@@ -147,12 +147,16 @@ func (client *Client) ShowVolumes(volume string) ([]common.VolumeObject, *common
...
 	} else {
 		logger.V(4).Info("-- ShowVolumesNamesGet", "status", httpRes.Status, "err", err, "body", httpRes.Body)
+		return nil, &common.ResponseStatus{}, err
 	}
 
 	logger.V(4).Info("================================================================================")
 
 	returnVolumes := []common.VolumeObject{}
-	status := CreateCommonStatus(logger, &response.Status)
+	status := &common.ResponseStatus{}
```
Comment (jskazinski) [inline pkg/api/volumes.go:156]: If response.Status is nil we return a default common.ResponseStatus(). Please look into either a) setting some values for common.ResponseStatus(), or b) changing CreateCommonStatus to handle a nil response.Status and filling in the correct common.ResponseStatus() to be returned to the caller.

**Review by jskazinski — CHANGES_REQUESTED** [review]
I have one question regarding the status returned from ShowVolumes() and one small cut-n-paste to correct.

File: pkg/api/mcapi.go [hunk 3]
```diff
@@ -96,11 +105,11 @@ func (client *Client) SessionValid(addr, username string) bool {
 			logger.V(2).Info("session invalid", "sessionkey", client.SessionKey)
 			return false
 		}
-		logger.V(2).Info("client is configured", "ipaddress", addr)
-		return true
 	}
 
-	return false
+	//run an API call to test that the session is still valid
+	_, _, err := client.apiClient.DefaultApi.ShowSystemGet(client.Ctx).Execute()
```
[the remark this answers is not in the document]
Comment (David-T-White) [reply pkg/api/mcapi.go]: Changed to use show controller-date which shows improved performance compared to show system.

**Review by jskazinski — APPROVED** [review]
Looks good, thanks David!

**Pull request merged**